src/sentry/search/events/base.py

```diff
@@ -41,3 +71,53 @@ def project_slugs(self) -> Mapping[str, int]:
 
     def column(self, name: str) -> Column:
         return Column(self.resolve_column_name(name))
+
+    def is_field_alias(self, alias: str) -> bool:
+        return alias in self.field_alias_converter
+
+    def resolve_field_alias(self, alias: str) -> SelectType:
```

Comment:
This return type is incompatible with `src/sentry/search/events/fields.py:1959` just gotta change 
`def resolve_field(self, field: str) -> Union[SnqlColumn, SnqlFunction]:`
to
`def resolve_field(self, field: str) -> SelectType:`